python/ray/rllib/common.py

```diff
@@ -1,15 +1,44 @@
 from collections import namedtuple
 from datetime import datetime
-import json
 import logging
 import os
+import simplejson
+import sys
 import tempfile
+import uuid
+try:
+    import smart_open
+except ImportError:
+    pass
+if sys.version_info[0] == 2:
+    import cStringIO as StringIO
+elif sys.version_info[0] == 3:
+    import io as StringIO
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
+class S3Logger(object):
+  """Writing small amounts of data to S3 with real-time updates.
+  """
+
+  def __init__(self, uri):
+    self.result_buffer = StringIO.StringIO()
```

Comment:
done in train.py (because in the algorithms it is used even before S3Logger is touched)